lib/node_runner_message_checker.go

```diff
@@ -0,0 +1,102 @@
+package sebak
+
+import (
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/error"
+	"boscoin.io/sebak/lib/network"
+	"boscoin.io/sebak/lib/node"
+)
+
+type NodeRunnerHandleMessageChecker struct {
+	sebakcommon.DefaultChecker
+
+	NodeRunner *NodeRunner
+	LocalNode  *sebaknode.LocalNode
+	NetworkID  []byte
+	Message    sebaknetwork.Message
+
+	Transaction Transaction
+}
+
+// CheckNodeRunnerHandleMessageTransactionUnmarshal makes `Transaction` from
+// incoming `sebaknetwork.Message`.
+func CheckNodeRunnerHandleMessageTransactionUnmarshal(c sebakcommon.Checker, args ...interface{}) (err error) {
+	checker := c.(*NodeRunnerHandleMessageChecker)
+
+	var tx Transaction
+	if tx, err = NewTransactionFromJSON(checker.Message.Data); err != nil {
+		return
+	}
+
+	if err = tx.IsWellFormed(checker.NetworkID); err != nil {
+		return
+	}
+
+	checker.Transaction = tx
+	checker.NodeRunner.Log().Debug("message is transaction")
+
+	return
+}
+
+// CheckNodeRunnerHandleMessageHasTransactionAlready checks transaction is in
```

Comment:
I don't think we need to repeat the name of the function here, esp. given it's quite long